numpy/core/getlimits.py

```diff
@@ -690,14 +690,16 @@ def __init__(self, int_type):
         self.kind = self.dtype.kind
         self.bits = self.dtype.itemsize * 8
         self.key = "%s%d" % (self.kind, self.bits)
-        if self.kind not in 'iu':
+        if self.kind not in 'ium':
             raise ValueError("Invalid integer data type %r." % (self.kind,))
 
     @property
     def min(self):
         """Minimum value of given dtype."""
         if self.kind == 'u':
             return 0
+        elif self.kind == 'm':
```

Comment:
So what exact is the point in accepting timedelta here if it only supports half of the `iinfo` interface anyway? It's questionable to what extent timedelta should be accepted here in the first place (as it's not an integer dtype), but with the previously mentioned limitations doesn't seem much more sensible to me than accepting floating or even non-numeric data types.

Reply:
You have a point there. But for some reason timedelta64 is specified as an integer data type elsewhere. Should we then leave this `iinfo` unmodified, but remove timedelta64 from the list of integer data types? Or do you have another idea?

Reply:
I clearly agree and I would need more convincing that there is much merit to it.  Should we have some of that info available?  Sure, but maybe we can tag it on as something else like `dtype.min_value` or so if really needed.

And yes, it shouldn't inherit from integer, maybe it is time to just change this (for 2.0) and try that out in the main branch.  There may or may not be some random/surprising fallout, but its worth a shot (I feel I had a start somewhere).

Reply:
> And yes, it shouldn't inherit from integer, maybe it is time to just change this (for 2.0) and try that out in the main branch. There may or may not be some random/surprising fallout, but its worth a shot (I feel I had a start somewhere).

This is something I'd personally be in favor of. There's a reason why in the typing stubs timedelta is not treated as a signedinteger subclass: the two are simply so radically different that it would be simply impossible to create decent type annotations. And this is not merely a typing-related problem: timedelta will pass `isinstance(..., np.integer)` checks during runtime despite it only duck-typing with the latter to a somewhat minimal extent.

Reply:
Thanks for your points and notes about the issue. I will look into how to modify timedelta not to inherit from integer.